src/libs/geotagging.c

```diff
@@ -774,89 +704,104 @@ void gui_init(dt_lib_module_t *self)
   d->timezones = _lib_geotagging_get_timezones();
   self->widget = gtk_box_new(GTK_ORIENTATION_VERTICAL, 0);
   dt_gui_add_help_link(self->widget, dt_get_help_url(self->plugin_name));
-  GtkBox *hbox;
-  GtkWidget *button;
-
-  /* the offset line */
-  hbox = GTK_BOX(gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0));
-
-  gtk_box_pack_start(hbox, dt_ui_label_new(_("time offset")), FALSE, TRUE, 0);
-
-  d->offset_entry = gtk_entry_new();
-  dt_gui_key_accel_block_on_focus_connect(d->offset_entry);
-  gtk_entry_set_max_length(GTK_ENTRY(d->offset_entry), 9);
-  gtk_entry_set_width_chars(GTK_ENTRY(d->offset_entry), 6);
-  gtk_box_pack_start(hbox, d->offset_entry, TRUE, TRUE, 0);
-  g_signal_connect(d->offset_entry, "key-press-event", G_CALLBACK(_lib_geotagging_offset_key_press), self);
-  g_signal_connect(d->offset_entry, "focus-out-event", G_CALLBACK(_lib_geotagging_offset_focus_out), self);
-  gtk_widget_set_tooltip_text(d->offset_entry, _("time offset\nformat: [+-]?[[hh:]mm:]ss"));
-  gchar *str = dt_conf_get_string("plugins/lighttable/geotagging/offset");
-  if(_lib_geotagging_parse_offset(str, NULL))
-    gtk_entry_set_text(GTK_ENTRY(d->offset_entry), str);
-  else
-    gtk_entry_set_text(GTK_ENTRY(d->offset_entry), "+00:00:00");
-  g_free(str);
 
-  GtkBox *button_box = GTK_BOX(gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0));
-  button = dtgtk_button_new(dtgtk_cairo_paint_zoom, CPF_STYLE_FLAT, NULL);
-  gtk_widget_set_tooltip_text(button, _("calculate the time offset from an image"));
-  gtk_box_pack_start(button_box, button, TRUE, TRUE, 0);
-  g_signal_connect(G_OBJECT(button), "clicked", G_CALLBACK(_lib_geotagging_show_offset_window), self);
+  GtkGrid *grid = GTK_GRID(gtk_grid_new());
+  gtk_grid_set_column_spacing(grid, DT_PIXEL_APPLY_DPI(5));
+  int line = 0;
+
+  GtkWidget *label = gtk_label_new(_("date time"));
+  gtk_widget_set_hexpand(label, TRUE);
+  gtk_widget_set_halign(label, GTK_ALIGN_START);
+  gtk_label_set_ellipsize(GTK_LABEL(label), PANGO_ELLIPSIZE_END);
+  gtk_grid_attach(grid, label, 0, line, 1, 1);
+  gtk_widget_set_tooltip_text(label, _("enter the new date time"
+                                       "\nkey in the new numbers or scroll over the cell"));
+
+  GtkWidget *box = _gui_init_datetime(&d->dt, 0, self);
+  gtk_widget_set_halign(box, GTK_ALIGN_END);
+  gtk_widget_set_hexpand(box, TRUE);
+  gtk_grid_attach(grid, box, 1, line++, 1, 1);
+
+  label = gtk_label_new(_("original date time"));
+  gtk_widget_set_hexpand(label, TRUE);
+  gtk_widget_set_halign(label, GTK_ALIGN_START);
+  gtk_label_set_ellipsize(GTK_LABEL(label), PANGO_ELLIPSIZE_END);
+  gtk_grid_attach(grid, label, 0, line, 1, 1);
+
+  box = _gui_init_datetime(&d->dt0, 1, self);
+  gtk_widget_set_halign(box, GTK_ALIGN_END);
+  gtk_widget_set_hexpand(box, TRUE);
+  gtk_grid_attach(grid, box, 1, line++, 1, 1);
+
+  label = gtk_label_new(_("date time offset"));
```

Comment:
I would add a tooltip saying that the format is : `dd hh:mm:ss`
It was not immediately clear to me until I change the month above and got 30 displayed.

Reply:
OK, I'll do.